ProxyAgent/HttpClient/HttpClient.cs

```diff
@@ -220,11 +225,24 @@ public HttpClient(ILogger logger)
             }
         }
 
-        private static void SetServerSslSecurity(IHttpRequest request, HttpClientHandler clientHandler)
+        private void SetServerSslSecurity(IHttpRequest request, HttpClientHandler clientHandler)
         {
             if (request.Options.AllowInsecureSslServer && request.UsesSsl())
             {
-                clientHandler.ServerCertificateCustomValidationCallback = delegate { return true; };
+                clientHandler.ServerCertificateCustomValidationCallback += delegate(HttpRequestMessage sender, X509Certificate2 cert, X509Chain chain, SslPolicyErrors error)
+                {
+                    var sslThumbprint = cert.Thumbprint.ToLowerInvariant();
+                    var configThumbprint = this.config.Thumbprint.ToLowerInvariant();
```

Comment:
nit: trim the string, that hit me several times when copying thumbprints from Windows certificate store